### ISSUE TITLE
Razor Components - onclick handler not called if href is set in <a> tag

### ISSUE THREAD

**Issue by MarkSFrancis** [aspnet/AspNetCore#9773]
### Describe the bug
Href attributes take precedence over click handlers in razor components for `<a>` tags

### To Reproduce
1. Create a new blazor app (server or client side - happens on both)
1. Edit the `Pages/Counter.razor` file's button
from
```
<button class="btn btn-primary" onclick="@IncrementCount">Click me</button>
```
to
```
<a href="#" class="btn btn-primary" onclick="@IncrementCount">Click me</a>
```
1. Instead of incrementing the counter, it will now jump to the "#" url, navigating to the home page

### Expected behavior
Increment the counter as it did before

### Additional context
Installed dotnet version:
dotnet 3.0.100-preview4-011223

**Comment by enetstudio** [comment]
I believe this is the default behavior of opening a new page, and, since you've got no such url, the default behavior of the Blazor routing system is to navigate to the home page.

Try to return false from your click event.
Try either of these:
`<a href="#" class="btn btn-primary" onclick="@IncrementCount;return false">Click me</a>`

`<a href="#" class="btn btn-primary" onclick="@(() => IncrementCount());return false">Click me</a>`

Hope this works...

**Comment by MarkSFrancis** [comment]
Unfortunately, it fails because it doesn't call the handler at all.
You could test this by writing a `Console.WriteLine()` in the handler and preserving the log: the handler is never called

**Comment by rjmr97** [comment]
According to this https://github.com/aspnet/Blazor/pull/937 pull request, .preventDefault() is no longer called when you click on a link.
> So it looks like we can fix #803 and all related issues trivially just be removing the event.preventDefault call. It doesn't appear to affect anything else negatively (...)


It is something you will be able to do in [3.0.0-preview6](https://github.com/aspnet/AspNetCore/milestone/39) according to this https://github.com/aspnet/AspNetCore/issues/5545 issue.

In the meantime, if it is possible in your situation, you can remove the `href="#"` from your tag and the event works again.

**Comment by MarkSFrancis** [comment]
Thanks @rjmr97, I'll close this ticket now as that one clearly covers this
